climpred/graphics.py

```diff
@@ -230,3 +236,194 @@ def plot_bootstrapped_skill_over_leadyear(bootstrapped, plot_persistence=True, a
+                lw=3,
+                ls=linestyles[i],
+                label=f'reference: {obs_name}',
+                zorder=hind.lead.size + 1,
+            )
+
+    # show only one item per label in legend
+    handles, labels = ax.get_legend_handles_labels()
+    by_label = OrderedDict(zip(labels, handles))
+    ax.legend(
+        by_label.values(), by_label.keys(), loc='center left', bbox_to_anchor=(1, 0.5)
+    )
+    ax.set_title('')
+    return ax
+
+
+def plot_ensemble_perfect_model(
+    pm, variable=None, ax=False, show_members=True, cmap='tab10'
+):
+    """Plot datasets from PerfectModelEnsemble.
+
+    Args:
+        pm (PredictionEnsemble): PredictionEnsemble.
+        variable (str or None): `variable` to plot. Defaults to the first in data_vars.
+        ax (plt.axes): axis to plot to. Defaults to False. Creates new.
+        show_members (bool): whether to display all members individually.
+            Defaults to True.
+        cmap (str): Name of plt colorbar. Defaults to 'tab10'.
+
+    Returns:
+        ax: plt.axes
+
+    """
+
+    check_only_climpred_dims(pm)
+    if not variable:
+        variable = list(pm._datasets['initialized'].data_vars)[0]
+    initialized = pm._datasets['initialized'][variable]
+    control = pm._datasets['control']
+    if isinstance(control, xr.Dataset):
+        control = control[variable]
+    calendar = infer_calendar_name(initialized.init)
+    lead_freq = get_lead_cftime_shift_args(initialized.lead.attrs['units'], 1)[1]
+
+    if not ax:
+        _, ax = plt.subplots(figsize=(10, 4))
+
+    cmap = mpl.cm.get_cmap(cmap, initialized.init.size)
+
+    for ii, i in enumerate(initialized.init.values):
+        dsi = initialized.sel(init=i).rename({'lead': 'time'})
+        # convert lead time into cftime
+        start_str = i.strftime()[:10]
+        # yi,mi,di = i.strftime().split(' ')[0].split('-')
+        # i_cfdatetime = cftime.datetime(int(yi),int(mi), int(di))
+        # try to plot vertical bars for each init
+        # now fails todo: how to axvline(cftime)?
+        # ax.axvline(x=i_cfdatetime,calendar='noleap')
+        # ax.axvline(x=int(init.init.isel(init=[i]).dt.year.values))
+        if initialized.lead.min() == 0:
+            dsi['time'] = xr.cftime_range(
+                start=start_str,
+                freq=lead_freq,
+                periods=dsi.time.size,
+                calendar=calendar,
+            )
+        elif initialized.lead.min() == 1:
+            dsi['time'] = xr.cftime_range(
+                start=start_str,
+                freq=lead_freq,
+                periods=dsi.time.size,
+                calendar=calendar,
+            )
+            # need to align with discussion about init times:
+            # I propose: initialization: as the exact date of the start of the forecast
+            #            observation/hist time: as the middle of time spanned
+            #            lead: as ints but when converted to real time as middle of time
```

Comment:
I would move this to the issue thread on this, and make a note for now here that this is how you're handling it

Reply:
moved to https://github.com/bradyrx/climpred/issues/304 where this belongs